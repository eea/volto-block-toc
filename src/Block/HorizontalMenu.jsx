/**
 * View toc block.
 * @module components/manage/Blocks/ToC/View
 */

import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import { map } from 'lodash';
import { Menu, Sticky } from 'semantic-ui-react';
import { FormattedMessage, injectIntl } from 'react-intl';
import AnchorLink from '@eeacms/volto-block-toc/AnchorLink';
// import { StickyContainer, Sticky } from 'react-sticky';

const RenderMenuItems = ({ items }) => {
  // console.log('items', items);
  return map(items, (item) => {
    const { id, level, title } = item;
    console.log('id, level, items', id, level, title);
    return (
      item && (
        <React.Fragment key={id}>
          <Menu.Item className={`headline-${level}`}>
            <AnchorLink href={`#${id}`}>{title}</AnchorLink>
          </Menu.Item>
          {item.items?.length > 0 && <RenderMenuItems items={item.items} />}
        </React.Fragment>
      )
    );
  });
};

/**
 * View toc block class.
 * @class View
 * @extends Component
 */
const View = ({ data, tocEntries }) => {
  console.log('toc entries', tocEntries);
  const stickyRef = useRef(null);
  return (
    <>
      {data.title && !data.hide_title ? (
        <h2>
          {data.title || (
            <FormattedMessage
              id="Table of Contents"
              defaultMessage="Table of Contents"
            />
          )}
        </h2>
      ) : (
        ''
      )}
      <Sticky
        onUnstick={(e) => console.log('HERE', e)}
        ref={stickyRef}
        offset={40}
      >
        <Menu>
          <RenderMenuItems items={tocEntries} />
        </Menu>
      </Sticky>
    </>
  );
};

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  properties: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default injectIntl(View);
