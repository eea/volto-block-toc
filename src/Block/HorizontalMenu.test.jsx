import React from 'react';
import renderer from 'react-test-renderer';
import HorizontalMenu from './HorizontalMenu';

jest.mock('react-intl', () => ({
  injectIntl: jest.fn((Component) => Component),
}));

describe('HorizontalMenu', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <HorizontalMenu properties={{}} data={{ title: 'Heading title' }} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
    expect(component).toMatchSnapshot();
  });
});
