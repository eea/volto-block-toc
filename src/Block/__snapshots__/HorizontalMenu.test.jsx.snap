// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalMenu renders correctly 1`] = `
Array [
  <h2>
    Heading title
  </h2>,
  <div
    className=""
  >
    <div
      style={Object {}}
    />
    <div
      className="ui sticky"
    >
      <div
        className="ui menu"
      />
    </div>
  </div>,
]
`;

exports[`HorizontalMenu renders correctly 2`] = `
Array [
  <h2>
    Heading title
  </h2>,
  <div
    className=""
  >
    <div
      style={Object {}}
    />
    <div
      className="ui sticky"
    >
      <div
        className="ui menu"
      />
    </div>
  </div>,
]
`;
